### PR TITLE
Automate "Data Driven" migrations?

### PR DESCRIPTION
**Pull request by zvecr**
## Description

```bash
$ qmk migrate -kb zvecr/split_blackpill -f VENDOR_ID -f PRODUCT_ID
Ψ Migrating keyboard zvecr/split_blackpill.
Ψ   Migrating file keyboards/zvecr/split_blackpill/config.h
Ψ     Migrating PRODUCT_ID...
Ψ     Migrating VENDOR_ID...
Ψ   Migrating file keyboards/zvecr/split_blackpill/rules.mk
Ψ   Updating keyboards/zvecr/split_blackpill/info.json
Ψ Migration of keyboard zvecr/split_blackpill complete!
Ψ Verify build with qmk compile -kb zvecr/split_blackpill -km default
```

Might be of use for the current USB config -> DD efforts.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fauxpark** [comment]
How disruptive is this to the rest of the `info.json`? They often have varying or mixed indentation, and may not follow the rules of `InfoJSONEncoder`. I'm aiming to minimise the diffs for ease of review, and the simplest (but obviously not the fastest) workflow I've come up with is to just make some snippets in Sublime and do it one board at a time.

**Comment by zvecr** [comment]
It is more disruptive, mostly in the area of layout macros. Next iteration I plan to add will be for layout macros, so bulk changes in that area wont matter as much. Nothing will stop manual migrations, however an automated process potentially needs less review.

Longer term, `info.json` formatting is something that I want to add to PR validation.

**Comment by tzarc** [comment]
bad stalebot

File: lib/python/qmk/cli/migrate.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    # Start migration
+    for file in files:
+        cli.log.info(f'  Migrating file {file}')
+        file_contents = file.read_text(encoding='utf-8').split('\n')
+        for key in keys:
+            for num, line in enumerate(file_contents):
+                if line.startswith(f'{key} =') or line.startswith(f'#define {key} '):
+                    cli.log.info(f'    Migrating {key}...')
+                    file_contents.pop(num)
```
Comment (tzarc) [inline lib/python/qmk/cli/migrate.py:64]:
```suggestion
                    while line.rstrip().endswith('\\'):
                        file_contents.pop(num)
                        line = file_contents[num]
                    file_contents.pop(num)
```

**Review by tzarc — CHANGES_REQUESTED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**